query-node/mappings/src/workingGroups.ts

```diff
@@ -739,6 +749,22 @@ export async function workingGroups_LeadRemarked({ store, event }: EventContext
     member.metadata.isVerifiedValidator = isVerified
     await store.save<MemberMetadata>(member.metadata)
     await store.save<Membership>(member)
+  } else if (metadata?.createTag && group.name === 'forumWorkingGroup') {
+    return await processCreateTag(store, metadata?.createTag, true)
+  } else if (metadata?.updateTag && group.name === 'forumWorkingGroup') {
+    return await processUpdateTag(store, metadata?.updateTag, true)
+  } else if (metadata?.assignTagsToThread && group.name === 'forumWorkingGroup') {
+    return await processAssignTagsToThread(store, metadata?.assignTagsToThread, true, 0)
+  } else if (metadata?.assignTagsToProposal && group.name === 'forumWorkingGroup') {
+    return await processAssignTagsToProposal(store, metadata?.assignTagsToProposal, true, 0)
+  } else if (metadata?.unassignTagsFromThread && group.name === 'forumWorkingGroup') {
+    return await processUnassignTagsFromThread(store, metadata?.unassignTagsFromThread, true, 0)
+  } else if (metadata?.unassignTagsFromProposal && group.name === 'forumWorkingGroup') {
+    return await processUnassignTagsFromProposal(store, metadata?.unassignTagsFromProposal, true, 0)
+  } else if (metadata?.allowTagToWorker && group.name === 'forumWorkingGroup') {
+    return await processAllowTagToWorker(store, metadata?.allowTagToWorker)
+  } else if (metadata?.disallowTagToWorker && group.name === 'forumWorkingGroup') {
+    return await processDisallowTagToWorker(store, metadata?.disallowTagToWorker)
```

Comment:
Let's extract those to a function as suggested earlier so the logic is not repeated. Additionally, with the current conditions if somebody from outside forum working group would try to work with tags, the message would just be ignored. Instead, it should return `invalidMetadata` error as is done above:
```
    if (group.name !== 'membershipWorkingGroup') {
      return invalidMetadata(`The ${group.name} can't verify the validator's membership`)
    }
```
I think what makes most sense here:
1. Replace all the conditions (same for regular worker message) with single:
```ts
else if (metadata?.createTag || metadata?.updateTag || ...) {
  return await processTagMessage(metadata, group.name, isLead, workerId)
}
```
2. In `processTagMessage` do:
  2.1. Check working group and return error if not forum
  2.2. Do permissions check so we don't have to duplicate across specific message handlers
  2.3. Have this long if-else for specific messages

WDYT?

Reply:
Also, please don't use `0` as worker ID for lead, that's misleading. If it's not needed for lead, just make it an optional parameter and skip